### PR TITLE
Update git.py notes

### PR DESCRIPTION
**Pull request by hyperized**
##### ISSUE TYPE
- Docs Pull Request
##### COMPONENT NAME

source_control/git
##### ANSIBLE VERSION

```
n/a
```
##### SUMMARY

Added notes section about SSH-agents that can be used to avoid unnecessary password prompts.

### CODE REVIEW

File: source_control/git.py [hunk 1]
```diff
@@ -190,6 +190,7 @@
       one solution is to use the option accept_hostkey. Another solution is to
       add the remote host public key in C(/etc/ssh/ssh_known_hosts) before calling
       the git module, with the following command: ssh-keyscan -H remote_host.com >> /etc/ssh/ssh_known_hosts."
+    - "When working with git over SSH, it's recommended you use your SSH agent to store the keys and passwords in order for the module not to prompt the key password several times during execution"
```
Comment (jjshoe) [inline source_control/git.py]: SSH agents don't store passwords.

Comment (jjshoe) [reply source_control/git.py]: Also, grammar could stand a significant re-work

**Review by jjshoe — CHANGES_REQUESTED** [review]

**Comment by gregdek** [comment]
Thanks @hyperized for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @hyperized for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**